src/turno.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * turno.cpp
+ *
+ *  Created on: 27 oct. 2018
+ *      Author: nazareno
+ */
+
+#include<turno.h>
+
+Turno::Turno(Tablero* tableroAsociado){
+	this->celulasMuertasTurno = 0;
+	this->celulasNacidasTurno = 0;
+	this->numeroDeTurno = 0; //capaz que lo ponemos en 1
+	this->tableroAsociado = tableroAsociado;
+}
+
+void Turno::marcarCambiosARealizarParaSiguienteTurno(){
+	for(int fila=0; fila < this->tableroAsociado->getFilas(); fila++){
+		for(int columna=0; columna < this->tableroAsociado->getColumnas(); columna++){
+			RGB* coloresCelulasVivasCircundantes [3];
+			int celulasCircundantesVivas = chequearCelulasCircundantes(fila, columna,coloresCelulasVivasCircundantes);
+			decidirVidaOMuerte(celulasCircundantesVivas, this->tableroAsociado->getParcela(fila,columna).getCoordenadaParcela(), coloresCelulasVivasCircundantes);
+		}
+	}
+}
+
+int Turno::chequearCelulasCircundantes(int fila, int columna, RGB* coloresCelulasVivasCircundantes[]){
+	int celulasCircundantesVivas;
+	if(this->tableroAsociado->getParcela(fila, columna).getCelula()->getEstado()){
```

Comment:
Se supone que getEstado() devuelve `true` si esta viva no? entonces porque si esta viva se achica el contador de `celulasCircundantesVivas`?
Por otro lado `celulasCircundantesVivas` no está inicializado por lo tanto tiene basura al principio, así que se estaría reduciendo cualquier numero.

Reply:
Tenes razon en el hecho de que cuando la variable no fue inicializada no tiene sentido restar.
De todas formas la resta se hace porque la iteracion pasa no solo por las celulas circundantes sino tambien por la celula en cuestion y si esta está viva suma uno a la cantidad de celulas circundantes

Reply:
Ahh ok, entonces podrías iterar algo así para no estar restando a mano, también creo que en el for interno deberías iterar por columna en vez de fila:
Podrías hacer algo así:
```
for (unsigned int i = fila - 1; i <= fila + 1; i++) {
		for (unsigned int j = columna - 1; j <= columna + 1; j++) {
			if (i >= 0 && j >= 0 && i < MAX_FILAS && j <  MAX_COLUMNAS) {// con esto prevenimos que salga del rango de la matriz
				bool mismoElemento = (i == fila && j == columna);
				if (mismoElemento == false && campo[i][j] == VIVA) {
					celulasCircundantesVivas++;
				}
			}
		}
	}
```